src/hotspot/cpu/x86/gc/z/zBarrierSetAssembler_x86.cpp

```diff
@@ -109,16 +109,16 @@ class ZRuntimeCallSpill {
       __ push2p(r30, r31);
     } else {
       if (_result != rax) {
-        __ push(rax);
+        __ paired_push(rax);
       }
-      __ push(rcx);
-      __ push(rdx);
-      __ push(rdi);
-      __ push(rsi);
-      __ push(r8);
-      __ push(r9);
-      __ push(r10);
-      __ push(r11);
+      __ paired_push(rcx);
```

Comment:
Hi @vamsi-parasa , for consecutive push/pop we can use push2/pop2 and 16byte alignment can be  guaranteed using following technique 
https://github.com/openjdk/jdk/pull/25351/files#diff-d5d721ebf93346ba66e81257e4f6c5e6268d59774313c61e97353c0dfbf686a5R94

Reply:
I like the current approach, unless we can come up with a very 'visually-low-overhead' way of adding extra alignment (i.e. this current change is across quite a few files, rather not complicate it). At most, perhaps something like `MacroAssembler::push_align()/MacroAssembler::pop_align()`, but I really rather not add more to this PR; It touches quite a few places so I like it being simpler.

As it stands, if nothing else, its clear from the `if` statement that existing path is left unmodified.

Reply:
Hi @vamsi-parasa , I think it's ok not to expose pop_ppx / push_ppx as separate interfaces, and let processor forward the values b/w push and matching pop if balancing constraints are satisfied.

Reply:
Hi Jatin (@jatin-bhateja), the reason to make the push_ppx/pop_ppx usage explicit is because an unbalanced push_ppx operation has a performance penalty.

Reply:
> Please create a new RFE for its tracking.

Please see the JBS issue (https://bugs.openjdk.org/browse/JDK-8362903) for push2/pop2 enabling in future.

Reply:
> Hi Jatin (@jatin-bhateja), the reason to make the push_ppx/pop_ppx usage explicit is because an unbalanced push_ppx operation has a performance penalty.

Thanks @vamsi-parasa , as per APX specification PPX is an optimization hint and should only improve performance if balancing contraintins are met. so I don't think it will have any performance penalty.